src/Chapter3.hs

```diff
@@ -910,6 +1011,21 @@ Implement instances of "Append" for the following types:
 class Append a where
     append :: a -> a -> a
 
+newtype Gold = Gold {
+    unGold :: Int
+  }
+instance Append Gold where
+    append :: Gold -> Gold -> Gold
+    append g h = Gold ((unGold g) + (unGold h))
+
+instance Append [a] where
+    append :: [a] -> [a] -> [a]
+    append g h = g ++ h
+
+instance (Append a) => Append (Maybe a) where
+    append :: Maybe a -> Maybe a -> Maybe a
+    append (Just g) (Just h) = Just (append g h)
+    append _ _ = Nothing
```

Comment:
While this is valid implementation, in reality, it would be more beneficial and it would make more sense if this function returned `Just` when at least one argument is `Just`. This is how `Maybe` works in the standard instance for a similar typeclass.